### PR TITLE
Feature/content type and more

### PR DESCRIPTION
**Pull request by shamasis**
This PR modularises the SDK to Node request options transformation functions. Post that, it augments only the form-data parser to support contentType + other options without disturbing the code path of `form` data mode.

Previously the same function was handling both.

TODO:

- [ ] release SDK and update package.json here
- [x] add unit tests to the request core spec

### CODE REVIEW

File: test/fixtures/upload-csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+key,value
```
Comment (kunagpal) [inline test/fixtures/upload-csv]: @codenirvana This file should have a CSV extension

File: lib/requester/core-body-builder.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    // if data has a truthy content type, we mutate the value to take the options. we are assuming that
+    // blank string will not be considered as an accepted content type.
+    if (param.contentType && _.isString(param.contentType)) {
+        (options || (options = {})).contentType = param.contentType;
+    }
+
+    // additionally parse the file name and length if sent
+    // @note: The filepath property overrides filename and may contain a relative path.
+    if (_.isString(param.fileName)) { (options || (options = {})).filename = param.fileName; }
```
[the remark this answers is not in the document]
Comment (shamasis) [reply lib/requester/core-body-builder.js]: We are not making a mistake to have consistency with someone else’ mistake!

**Comment by codenirvana** [comment]
@kunagpal There was a pending Todo. Need to update `package.json` with the updated SDK version.

**Comment by kunagpal** [comment]
@codenirvana That will be taken care of by GreenKeeper when the release has been created.

**Pull request merged**